server/middlewares/csp.js

```diff
@@ -4,9 +4,13 @@ let trusted = [
   "'self'"
 ];
 
+const host = process.env.HOST || 'localhost';
+const port = process.env.PORT || process.env.SYNC_PORT || '3000';
+
 if (process.env.NODE_ENV !== 'production') {
   trusted = trusted.concat([
-    'ws://localhost:3000'
+    `http://${host}:${port}`,
```

Comment:
I need one confirmation, while `self` would be able to cover this when the host is `localhost` it will not in be the same in case of most cloud IDEs.

C9's editor / terminal are available at an URL like:
`https://ide.c9.io/<username>/<workspace-name>`

While in execution context the app is only available at:
`https://<worspace-name>-<username>.c9users.io`

I think this is the reason, I had to add this in the first place to get past the errors that I got client side.

I'll re-confirm this soon. Thanks for the comments and QA